File: test/parse_test.dart
import 'package:calc_o_pad/parse.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test("parses simple addition", () {
    expect(parse("1 + 2"), const Add([Number(1), Number(2)]));
  });

  test("parses simple subtraction with three numbers", () {
    expect(
        parse("5 - 2 - 1"),
        const Subtract([
          Subtract([Number(5.0), Number(2.0)]),
          Number(1.0)
        ]));
  });

  test("assigns to a variable", () {
    expect(parse("x: 1"), const Assign("x", Number(1)));
  });

  test("assigns a complex expression to a variable", () {
    expect(
        parse("x: 1 + 2 * 3"),
        const Assign(
            "x",
            Add([
              Number(1),
              Multiply([Number(2), Number(3)])
            ])));
  });

  test("parses a variable as a part of the expression", () {
    expect(parse("x + 1"), const Add([Variable("x"), Number(1)]));
  });
}

File: calc-o-pad-js/src/parse.test.ts
import test from "ava";
import { parse } from "./parse";

test("parses simple addition", (t) => {
  t.deepEqual(parse("1 + 2"), {
    type: "add",
    values: [
      { type: "number", value: 1 },
      { type: "number", value: 2 },
    ],
  });
});

test("parses simple subtraction with three numbers", (t) => {
  t.deepEqual(parse("5 - 2 - 1"), {
    type: "subtract",
    values: [
      { type: "number", value: 5 },
      { type: "number", value: 2 },
      { type: "number", value: 1 },
    ],
  });
});
